File: Astromaximum/Classes/utils/AmaxCommonDataFile.m
//
//  AmaxCommonDataFile.m
//  Astromaximum
//
//  Created by admin on 13.11.12.
//  Copyright (c) 2012 S&W Axis. All rights reserved.
//

#import "AmaxCommonDataFile.h"
#import "Astromaximum-Bridging-Header.h"
#import "Astromaximum-Swift.h"

@implementation AmaxCommonDataFile

@synthesize startYear = _startYear;
@synthesize startMonth = _startMonth;
@synthesize startDay = _startDay;
@synthesize dayCount = _dayCount;
@synthesize mData = _mData;

- (id)initWithFilePath:(NSString *)filePath
{
    const int isLegacy = false;
    NSData *fullData = [NSData dataWithContentsOfFile:filePath];
    AmaxDataInputStream *is = [[AmaxDataInputStream alloc]initWithData:fullData];
    _startYear = [is readShort];
    _startMonth = [is readUnsignedByte] - 1;
    _startDay = [is readUnsignedByte];
    
    Size customDataLen = [is readUnsignedShort]; // customData length
    if (isLegacy)
        _dayCount = [is readShort];
    else
        _monthCount = [is readUnsignedByte];
    if (customDataLen > 0) {
        customData = (NSMutableData*)[is readWithLength:customDataLen];
    }
    Size bufferLength = [is availableBytes];
    NSMutableData* buffer = (NSMutableData*)[is readWithLength:bufferLength];
    _mData = [[AmaxDataInputStream alloc]initWithData:buffer];
    return self;
}

@end

File: Astromaximum/Classes/utils/AmaxCommonDataFile.swift
//
//  AmaxCommonDataFile.swift
//  Astromaximum2
//
//  Created by Andrei Ivushkin on 12.07.2020.
//  Copyright © 2020 S&W Axis. All rights reserved.
//

import Foundation

class AmaxCommonDataFile {

    var startYear:Int = 0
    var startMonth:Int = 0
    var startDay:Int = 0
    var dayCount:Int = 0
    var monthCount:Int = 0
    var customData:Data?
    var data:AmaxDataInputStream?

    init(filePath: String) {
        let isLegacy = false;
        guard let fullData = (NSData(contentsOfFile: filePath) as Data?) else {
            print("AmaxCommonDataFile: failed to open \(filePath)")
            return;
        }
        let ais = AmaxDataInputStream(data: fullData)
        startYear = Int(ais.readShort())
        startMonth = Int(ais.readUnsignedByte())
        startDay = Int(ais.readUnsignedByte())

        let customDataLen = Int(ais.readUnsignedShort()) // customData length
        if isLegacy {
            dayCount = Int(ais.readShort())
        }
        else {
            monthCount = Int(ais.readUnsignedByte())
        }
        if customDataLen > 0 {
            customData = ais.read(length: customDataLen)
        }
        let bufferLength = ais.availableBytes()
        data = AmaxDataInputStream(data: ais.read(length: bufferLength))
    }
}
